### ISSUE TITLE
`dotnet run` does not work with .NET Framework on Linux

### ISSUE THREAD

**Issue by bradwilson** [dotnet/sdk#11787]
When trying to run a program that was built for .NET Framework on Linux, I am able to run the program directly via `mono program.exe`, but I cannot run the program through .NET SDK CLI `dotnet run`.

#### dotnet-run-repro.csproj

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFrameworks>net472;netcoreapp3.1</TargetFrameworks>
  </PropertyGroup>

  <ItemGroup Condition=" '$(TargetFrameworkIdentifier)' == '.NETFramework' ">
    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0" />
  </ItemGroup>

</Project>
```

#### Program.cs

```csharp
using System;

namespace dotnet_run_repro
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello World!");
        }
    }
}
```

### Repro steps

```
  sh$ dotnet build
Microsoft (R) Build Engine version 16.6.0+5ff7b0c9e for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  dotnet-run-repro -> /home/bradwilson/dev/dotnet-run-repro/bin/Debug/net472/dotnet-run-repro.exe
  dotnet-run-repro -> /home/bradwilson/dev/dotnet-run-repro/bin/Debug/netcoreapp3.1/dotnet-run-repro.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.33

  sh$ dotnet run --framework netcoreapp3.1
Hello World!

  sh$ dotnet run --framework net472
System.ComponentModel.Win32Exception (13): Permission denied
   at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at Microsoft.DotNet.Cli.Utils.Command.Execute()
   at Microsoft.DotNet.Tools.Run.RunCommand.Execute()
   at Microsoft.DotNet.Tools.Run.RunCommand.Run(String[] args)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)

  sh$ mono bin/Debug/net472/dotnet-run-repro.exe
Hello World!
```

### Environment

```
  sh$ dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.300
 Commit:    b2475c1295

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  18.04
 OS Platform: Linux
 RID:         ubuntu.18.04-x64
 Base Path:   /usr/share/dotnet/sdk/3.1.300/

Host (useful for support):
  Version: 3.1.4
  Commit:  0090613580

.NET Core SDKs installed:
  2.1.806 [/usr/share/dotnet/sdk]
  3.1.300 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.18 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.18 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.4 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.18 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.4 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

  sh$ mono --version
Mono JIT compiler version 6.8.0.123 (tarball Tue May 12 15:11:57 UTC 2020)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug
	Interpreter:   yes
	LLVM:          yes(610)
	Suspend:       hybrid
	GC:            sgen (concurrent by default)
```

**Comment by dsplaisted** [comment]
This would be low priority for us, but we'd be willing to consider a PR for it.  It would likely involve updating how `RunCommand` is calculated in Microsoft.NET.Sdk.targets: https://github.com/dotnet/sdk/blob/1e3d5e59764d86c2ab5a6815276f3cd0be0968fe/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L762-L767

**Comment by bradwilson** [comment]
@dsplaisted I could certainly redo it to launch via `mono`, but how is that impacted by .NET 5? Will launching executables on .NET 5 on non-Windows machines still be done by invoking `mono`?

**Comment by dsplaisted** [comment]
No, .NET 5 is the next version of .NET Core, so it will run with the dotnet host the same as currently happens with .NET Core.  You would only want to run using `mono` if the project targets .NET 4.x or lower and the current OS is not Windows.

**Comment by eriawan** [comment]
@dsplaisted 
Pardon, I think this issue is fixed on the merged PR of #11883 
Could we close this issue? Or is there any more work left to do about this issue?
